rskj-core/src/main/java/co/rsk/peg/BridgeMethods.java

```diff
@@ -822,7 +822,20 @@ public enum BridgeMethods {
         activations -> activations.isActive(RSKIP271),
         fixedPermission(false),
         CallTypeHelper.ALLOW_STATIC_CALL
-    );
+    ),
+    // Union Bridge Methods
+    SET_UNION_BRIDGE_CONTRACT_ADDRESS_FOR_TESTNET(
+        CallTransaction.Function.fromSignature(
+            "setUnionBridgeContractAddressForTestnet",
+            new String[]{"address"},
+            new String[]{"int"}
+        ),
+        fixedCost(24000L),
+        (BridgeMethodExecutorTyped<Integer>) Bridge::setUnionBridgeContractAddressForTestnet,
+        activations -> activations.isActive(RSKIP502),
```

Comment:
Perhaps we could move this to a method that asserts both that RSKIP502 is active and that the newtork is not mainnet. See how it's done in RECEIVE_HEADERS for example

Reply:
I prefer the way it is now. I see the `Bridge.executeIfElse` too complex and overkill for this. I tried to implement it and could not make the logic of the RSKIP and the environment enable work. Let's talk about it more.

Reply:
Task created to address this proposal later.